src/renderer/NumberInput.tsx

```diff
@@ -47,65 +56,131 @@ const NumberInput = ({
+      }
+      if (temp !== 1 || floatPrecision === 1) {
+        throw new Error('Bad floatPrecision value.');
+      }
+      return factor;
     }
-    return value;
-  }, [value]);
+    return 1;
+  }, [floatPrecision, type]);
 
   const onInput = (e: ChangeEvent<HTMLInputElement>) => {
     const { value: input } = e.target;
-    const lastChar = input[input.length - 1];
 
     // Allow user to clear input and type an initial negative sign
-    if (input === '' || (input === '-' && min < 0)) {
+    if (
+      input === '' ||
+      (input === '-' && min < 0) ||
+      (input === '.' && type === 'float')
+    ) {
       setInternalValue(input);
       setHasChanges(true);
       return;
     }
 
-    // Prevent user from typing non-numerical values
-    if (Number.isNaN(parseInt(lastChar, 10))) {
-      return;
+    let num = NaN;
+    if (type === 'int') {
+      // If parseInt encounters a character that is not a numeral in the
+      // specified radix, it ignores it and all succeeding characters and
+      // returns the integer value parsed up to that point. parseInt truncates
+      // numbers to integer values. Leading and trailing spaces are allowed.
+      num = parseInt(input, 10);
+      if (num.toString() !== input) {
+        // illegal character in the input or number too large
+        return;
+      }
+    } else if (type === 'float') {
+      // If parseFloat encounters a character other than a plus sign (+),
+      // minus sign (- U+002D HYPHEN-MINUS), numeral (0–9), decimal point (.),
+      // or exponent (e or E), it returns the value up to that character,
+      // ignoring the invalid character and characters following it.
+      // We disallow e / E.
+      if (input.match(/e|E/)) {
+        return;
+      }
+      // parseFloat("10.00") => 10. Since a user might input 10.001, we must
+      // allow typing in "10.00". We'll append a 1 to avoid the truncation
+      // of 0s done by parseFloat and see if the input is valid.
+      const testInput = `${input}1`;
+      num = parseFloat(testInput);
+      if (num.toString() !== testInput) {
+        // illegal character in the input
+        return;
+      }
     }
 
-    const newValue: number = parseInt(input, 10);
-    // Prevent user from typing numbers with more than 5 digits
-    if (Math.abs(newValue) >= 100000) {
+    // Prevent user from typing numbers that are too large or use more than 7 characters
+    if (Math.abs(num) > 10000 || input.length > 7) {
       return;
     }
-    setHasChanges(input !== value);
-    setInternalValue(newValue);
+
+    setHasChanges(input !== value.toString());
+    setInternalValue(input);
   };
 
   // Helper for discarding changes
   const onBlur = () => {
     setHasChanges(false);
-    setInternalValue(value);
+    setInternalValue(value.toString());
   };
 
   const onArrow = async (isIncrement: boolean) => {
-    const offset = isIncrement ? 1 : -1;
+    const offset = isIncrement ? floatPrecision : floatPrecision * -1;
```

Comment:
Make sure this is correct after the refactor.

Currently this is a bug since the offset should be 1 if the type is an int